myFitnessTrainer/src/screens/AvatarScreen.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import { Button, View, StyleSheet, TouchableOpacity, Text } from 'react-native';
+import * as ImagePicker from 'expo-image-picker';
+import { getStorage, ref, uploadBytes, getDownloadURL } from 'firebase/storage';
+import { storage } from '../../firebaseConfig';
+import Avatar from '../components/Avatar';
+
+const pickedImageFileName = "avatar.png";
+const defaultAvatarFileName = "default.png";
+
+const AvatarScreen = ({navigation}) =>  {
+  const [defaultImageURL, setDefaultImageURL] = useState(null)
+  const [avatarURL, setAvatarURL] = useState(null)
+  const [image, setImage] = useState(null);
+
+  // get the default.png from Firebase. We'll use its URL as the User's Avatar URL if
+  // custom image is not picked from the user's device
+  const defaultImageRef = ref(storage, defaultAvatarFileName);
+  getDownloadURL(defaultImageRef)
+    .then((url) => {
+      setDefaultImageURL(url);
+      setAvatarURL(url);
+    });
+
+  const pickImage = async () => {
+    let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.Images,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 0,
+    });
+    if (!result.canceled) {
+      setImage(result.assets[0].uri);
+    }
+  };
+
+  const saveAvatarSelection = async () => {
+    if (image){ // if user picked an image from their device
+      // upload image to Firebase
+      // https://github.com/expo/examples/blob/master/with-firebase-storage-upload/App.js#L178-L205
+      const blob = await new Promise((resolve, reject) => {
+        const xhr = new XMLHttpRequest();
+        xhr.onload = function () {
+          resolve(xhr.response);
+        };
+        xhr.onerror = function (e) {
+          console.log(e);
+          reject(new TypeError("Network request failed"));
+        };
+        xhr.responseType = "blob";
+        xhr.open("GET", image, true);
+        xhr.send(null);
+      });
+      // save image from device as "avatar.png"
+      const fileRef = ref(getStorage(), pickedImageFileName);
+      await uploadBytes(fileRef, blob);
+      blob.close();
+
+      // get URL of avatar.png
+      const pickedImageRef = ref(storage, pickedImageFileName);
+      getDownloadURL(pickedImageRef)
+        .then((url) => {
+          // set user's avatar URL as the URL of avatar.png
+          setAvatarURL(url);
+        });
+      console.log("image was picked; avatarURL=", avatarURL)
+    
+    let newStorageLocation = "gs://" + pickedImageRef._location.bucket + "/" + pickedImageFileName;
+    // Ex: newStorageLocation = gs://myfitnesstrainer-94289.appspot.com/avatar.png
+    console.log("newStorageLocation=", newStorageLocation);
+    // TODO: implement functionality to update the User's avatars/avatar storageLocation 
```

Comment:
We had originally planned to make our own database collection called avatars (currently in there), but the more I think about it, the more I'm ok with just having an avatar_url in the users collection. 
When a user signs up, we'll set their avatar_url to the `defaultImageRef`'s url. Then we'll just replace it if they upload an image.

Reply:
If we agree on this, then this would be the correct location (as mentioned in the TODO), to add the `newStorageLocation `(url) to the `avatar_url `field of the user.

Reply:
I think we find the user's doc by finding the 

`currentUserEmail `= `auth.currentUser?.email`, then filtering the users collection like this:

(import db from firebaseconfig)

```
collection(db, "users")
.where("email", "==", currentUserEmail )

```

Reply:
Thanks for the code! It'll help save time. 
And I'll add another TODO to look into restructuring the user collection (looking into adding avatar_url attribute directly instead)